### PR TITLE
Tl/react v18

### PR DESCRIPTION
**Pull request by tjlav5**
Update the app to react18

Includes updates to react-testing-library/x

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -49,29 +49,32 @@ const RouterWithInit = () => {
...
 
-ReactDOM.unstable_createBlockingRoot(document.getElementById('root')!).render(
-  <React.StrictMode>
-    <RMWCProvider
-      // Globally disable ripples
-      ripple={false}
+const container = document.getElementById('root');
+const root = createRoot(container!);
+
+// TODO: Enable strict-mode after upgrading to router-v6
```
Comment (yuchenshi) [inline src/index.tsx:55]:
```suggestion
// TODO: Enable strict-mode after upgrading to router-v6 (or upcoming v5 backport)
```

**Review by yuchenshi — APPROVED** [review]
LGTM, a few seemingly behavior changes inline and a few nits

File: src/components/Firestore/DocumentEditor/index.test.tsx [hunk 2]
```diff
@@ -225,24 +224,22 @@ describe('changing types', () => {
...
+    result = await renderWithFirestore(async () => (
+      <DocumentEditor value={{ hello: 'world' }} />
+    ));
 
     // Wait for selects to be stable
     await new Promise((resolve) => setTimeout(resolve, 500));
 
     const { getByLabelText } = result;
-    setType = (fieldType: FieldType) =>
+    setType = async (fieldType: FieldType) =>
```
[the remark this answers is not in the document]
Comment (tjlav5) [reply src/components/Firestore/DocumentEditor/index.test.tsx:235]: Ah erroneous, removed

File: src/components/Firestore/DocumentEditor/index.test.tsx [hunk 3]
```diff
@@ -225,24 +224,22 @@ describe('changing types', () => {
...
       fireEvent.change(getByLabelText(/Type/), {
         target: { value: fieldType },
       });
   });
 
   it('switches to an array', async () => {
-    const { getByLabelText, getAllByText } = result;
+    const { getAllByText } = result;
     await act(async () => {
```
[the remark this answers is not in the document]
Comment (tjlav5) [reply src/components/Firestore/DocumentEditor/index.test.tsx:243]: Not sure, legacy? Removed though and tests are happy. Good catch

**Pull request merged**